api/v1beta2/network_types.go

```diff
@@ -27,53 +27,106 @@ type NetworkStatus struct {
+type ElbScheme string
 
 var (
-	// ClassicELBSchemeInternetFacing defines an internet-facing, publicly
-	// accessible AWS Classic ELB scheme.
-	ClassicELBSchemeInternetFacing = ClassicELBScheme("internet-facing")
+	// ElbSchemeInternetFacing defines an internet-facing, publicly
+	// accessible AWS ELB scheme.
+	ElbSchemeInternetFacing = ElbScheme("internet-facing")
 
-	// ClassicELBSchemeInternal defines an internal-only facing
+	// ElbSchemeInternal defines an internal-only facing
 	// load balancer internal to an ELB.
-	ClassicELBSchemeInternal = ClassicELBScheme("internal")
+	ElbSchemeInternal = ElbScheme("internal")
 
-	// ClassicELBSchemeIncorrectInternetFacing was inaccurately used to define an internet-facing LB in v0.6 releases > v0.6.6 and v0.7.0 release.
-	ClassicELBSchemeIncorrectInternetFacing = ClassicELBScheme("Internet-facing")
+	// ElbSchemeIncorrectInternetFacing was inaccurately used to define an internet-facing LB in v0.6 releases > v0.6.6 and v0.7.0 release.
+	ElbSchemeIncorrectInternetFacing = ElbScheme("Internet-facing")
 )
 
-func (e ClassicELBScheme) String() string {
+func (e ElbScheme) String() string {
 	return string(e)
 }
 
-// ClassicELBProtocol defines listener protocols for a classic load balancer.
-type ClassicELBProtocol string
+// ElbProtocol defines listener protocols for a load balancer.
+type ElbProtocol string
 
-func (e ClassicELBProtocol) String() string {
+func (e ElbProtocol) String() string {
 	return string(e)
 }
 
 var (
-	// ClassicELBProtocolTCP defines the ELB API string representing the TCP protocol.
-	ClassicELBProtocolTCP = ClassicELBProtocol("TCP")
+	// ElbProtocolTCP defines the ELB API string representing the TCP protocol.
+	ElbProtocolTCP = ElbProtocol("TCP")
 
-	// ClassicELBProtocolSSL defines the ELB API string representing the TLS protocol.
-	ClassicELBProtocolSSL = ClassicELBProtocol("SSL")
+	// ElbProtocolSSL defines the ELB API string representing the TLS protocol.
+	ElbProtocolSSL = ElbProtocol("SSL")
 
-	// ClassicELBProtocolHTTP defines the ELB API string representing the HTTP protocol at L7.
-	ClassicELBProtocolHTTP = ClassicELBProtocol("HTTP")
+	// ElbProtocolHTTP defines the ELB API string representing the HTTP protocol at L7.
+	ElbProtocolHTTP = ElbProtocol("HTTP")
 
-	// ClassicELBProtocolHTTPS defines the ELB API string representing the HTTP protocol at L7.
-	ClassicELBProtocolHTTPS = ClassicELBProtocol("HTTPS")
+	// ElbProtocolHTTPS defines the ELB API string representing the HTTP protocol at L7.
+	ElbProtocolHTTPS = ElbProtocol("HTTPS")
+
+	// ElbProtocolTLS defines the NLB API string representing the TLS protocol.
+	ElbProtocolTLS = ElbProtocol("TLS")
+	// ElbProtocolUDP defines the NLB API string representing the UPD protocol.
+	ElbProtocolUDP = ElbProtocol("UDP")
+)
+
+// TargetGroupHealthCheck defines health check settings for the target group.
+// TODO: Create default values for these.
+type TargetGroupHealthCheck struct {
```

Comment:
Good question! I think the target group has some defaults if not defined. I'll work it out and define some defaults. The problem is that not all load balancer types support all health check fields which is crazy. 🤣

Reply:
Yeah, since the target groups are defined separately, I can't add default values for these. The default Listener for NLB right now is implemented with default values in any case. If any more are added, it's the user's responsibility to fill them correctly.

That said, for NLB, the only fields used are `Protocol` and `Port` because the others aren't supported. So the other fields are ignored.

Reply:
Oh yeah, I'll add `omitempty`.